### PR TITLE
Maintenance/update deps

### LINKED ISSUE #504: Migrate floor_generator's dependencies to null-safe versions
We have recently migrated the codebase to support null-safety (#462). We're now generating null-safe code but floor_generator's internal dependencies aren't exclusively null-safe.

Migrate the dependencies to their null-safe versions once available and make sure to not run tests with `--no-sound-null-safety` anymore (CI).

- [x] build
- [x] build_config
- [x] build_test
- [x] code_builder
- [x] dart_style
- [x] source_gen

Additionally, make sure to get rid of the mockito (5.0.2) dependency override, which was introduced to make floor work with source_gen 1.0.0. The tracking issue for mockito can be found here: https://github.com/dart-lang/mockito/issues/373.

### PR DESCRIPTION
**Pull request by mqus**
Address #541 by updating the dependencies

Afaict this should be the last step of the null-safety migration, so I removed the `--no-sound-null-safety` flag from the CI and it also ran through successfully on my PC. 

Closes #504

### CODE REVIEW

**Comment by mqus** [comment]
(not sure which label to use though :D )

**Review by demos77 — APPROVED** [review]

File: docs/getting-started.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ dependencies:
 
 dev_dependencies:
   floor_generator: ^1.0.1
-  build_runner: ^1.12.2
+  build_runner: ^2.0.0
```
Comment (vitusortner) [inline docs/getting-started.md:20]: 💡 We could also update the `build_runner` versions in `README.md` and `floor/README.md`.

**Review by vitusortner — APPROVED** [review]
Thanks for keeping the project's dependencies up-to-date!
I'm currently looking into transferring the package to a verified publisher instead of relying on my personal email address for releases. It'll also simplify having multiple publishers so that all maintainers can create releases.

**Comment by mqus** [comment]
I also noticed that there are some files that still contain the 
```dart
// ignore_for_file: import_of_legacy_library_into_null_safe
```
header, which I can probably remove now

**Comment by mqus** [comment]
> Thanks for keeping the project's dependencies up-to-date!

Only because you made it so easy to do that :D

**Review by demos77 — APPROVED** [review]

**Review by vitusortner — APPROVED** [review]

**Pull request merged**